cmd/zoekt-sourcegraph-indexserver/cleanup.go

```diff
@@ -83,21 +121,12 @@ func cleanup(indexDir string, repos []uint32, now time.Time, shardMerging bool)
 			_ = os.Chtimes(shard.Path, now, now)
 		}
 
-		if shardMerging {
-			// 1 repo can be split across many simple shards but it should only be contained
-			// in 1 compound shard. Hence we check that len(shards)==1 and only consider the
-			// shard at index 0.
-			if len(shards) == 1 && strings.HasPrefix(filepath.Base(shards[0].Path), "compound-") {
-				shardsLog(indexDir, "tomb", shards)
-				if err := zoekt.SetTombstone(shards[0].Path, repo); err != nil {
-					log.Printf("error setting tombstone for %v in shard %s: %s. Removing shard\n", repo, shards[0].Path, err)
-					_ = os.Remove(shards[0].Path)
-				}
-				continue
-			}
+		if shardMerging && maybeSetTombstone(shards, repo) {
+			shardsLog(indexDir, "tomb", shards)
+			continue
 		}
 		moveAll(trashDir, shards)
-		shardsLog(indexDir, "remove", shards)
+		shardsLog(indexDir, "removename", shards)
```

Comment:
Shouldn't this stay "remove" so that we can distinguish between this and the case where we delete because of inconsistent names?

Reply:
whoops, you are right. don't know how I managed to do this, I was obviously doing too much refactoring at some point and this is the evidence.
```suggestion
		shardsLog(indexDir, "remove", shards)
```